dev/src/document.ts

```diff
@@ -382,12 +388,14 @@ export class DocumentSnapshot<T = DocumentData> {
 
     // We only want to use the converter and create a new QueryDocumentSnapshot
     // if a converter has been provided.
-    if (this.ref._converter !== defaultConverter) {
+    if (this.ref._converter !== defaultConverter()) {
       const untypedReference = new DocumentReference(
         this.ref.firestore,
         this.ref._path
       );
-      return this.ref._converter.fromFirestore(
+      return ((this.ref._converter as unknown) as FirestoreDataConverter<
```

Comment:
This compiles for me as just "this.ref._converter.fromFirestore".

Reply:
done.